rust/services/workload/src/host_api.rs

```diff
@@ -553,19 +559,33 @@ mod util {
                     ));
                 }
 
+                if enable_http_gw {
+                    // reminder: we pass the the workload_id as the installed_app_id at app install time
+                    // eventually these may more more than one
+                    let list_stringified = &[workload_id]
+                        .iter()
+                        .map(|s| format!(r#""{s}""#))
+                        .collect::<Vec<_>>()
+                        .join(" ");
+
+                    override_attrs.push(format!("httpGwAllowedAppIds = [{list_stringified}]"));
+                }
+
                 let override_attrs_stringified = override_attrs.join("; ") + ";";
 
                 log::debug!(
                     "passing the these overrides to holochain: '{override_attrs_stringified}'"
                 );
 
                 let nix_build_args = [
+                    "--refresh",
                     "--extra-experimental-features",
                     "nix-command flakes",
                     "--impure",
                     "--expr",
                     &format!(
-                        r#"(builtins.getFlake "github:holo-host/holo-host/host-agent-workloads-container").packages.${{builtins.currentSystem}}.extra-container-holochain.override {{ {override_attrs_stringified} }}"#
+                        // TODO(feat): make this configurable and use something more dynamic.
+                        r#"(builtins.getFlake "github:holo-host/holo-host/hc-http-gw-prototyping").packages.${{builtins.currentSystem}}.extra-container-holochain.override {{ {override_attrs_stringified} }}"#
```

Comment:
post-merge we shall apply this:

```suggestion
                        r#"(builtins.getFlake "github:holo-host/holo-host/main").packages.${{builtins.currentSystem}}.extra-container-holochain.override {{ {override_attrs_stringified} }}"#
```